File: Python/GameLife.py
import numpy as np
import os
import time

def print_board(board):
    os.system('cls')
    for row in board:
        print(''.join(row))
    print()

def count_neighbors(board, row, col):
    count = 0
    for i in range(row-1, row+2):
        for j in range(col-1, col+2):
            if i != row or j != col:
                count += board[(i % len(board)), (j % len(board[0]))] == '*'
    return count

def next_generation(board):
    new_board = np.copy(board)
    for i in range(len(board)):
        for j in range(len(board[0])):
            neighbors = count_neighbors(board, i, j)
            if board[i, j] == '*':
                if neighbors < 2 or neighbors > 3:
                    new_board[i, j] = ' '
            else:
                if neighbors == 3:
                    new_board[i, j] = '*'
    return new_board

# Tamaño del tablero
ROWS = 30
COLS = 30

# Inicializar tablero aleatoriamente
board = np.random.choice([' ', '*'], size=(ROWS, COLS))

# Ciclo principal del juego
while True:
    print_board(board)
    time.sleep(0.2)  # Pausa de 0.2 segundos entre generaciones
    board = next_generation(board)

File: JavaScript/GameLife.js
// Función para imprimir el tablero
function printBoard(board) {
    console.clear();
    for (let row of board) {
        console.log(row.join(''));
    }
    console.log();
}

// Función para contar vecinos vivos
function countNeighbors(board, row, col) {
    let count = 0;
    for (let i = row - 1; i <= row + 1; i++) {
        for (let j = col - 1; j <= col + 1; j++) {
            if (i !== row || j !== col) {
                if (board[i] && board[i][j] === '*') {
                    count++;
                }
            }
        }
    }
    return count;
}

// Función para calcular el siguiente estado del tablero
function nextGeneration(board) {
    const newBoard = [];
    for (let i = 0; i < board.length; i++) {
        const newRow = [];
        for (let j = 0; j < board[i].length; j++) {
            const neighbors = countNeighbors(board, i, j);
            if (board[i][j] === '*') {
                if (neighbors < 2 || neighbors > 3) {
                    newRow.push(' ');
                } else {
                    newRow.push('*');
                }
            } else {
                if (neighbors === 3) {
                    newRow.push('*');
                } else {
                    newRow.push(' ');
                }
            }
        }
        newBoard.push(newRow);
    }
    return newBoard;
}

// Tamaño del tablero
const ROWS = 10;
const COLS = 30;

// Inicializar tablero aleatoriamente
const board = Array.from({ length: ROWS }, () =>
    Array.from({ length: COLS }, () => (Math.random() > 0.5 ? '*' : ' '))
);

// Ciclo principal del juego
setInterval(() => {
    printBoard(board);
    const newBoard = nextGeneration(board);
    for (let i = 0; i < board.length; i++) {
        for (let j = 0; j < board[i].length; j++) {
            board[i][j] = newBoard[i][j];
        }
    }
}, 200);
